File: mario.c
#include <cs50.h>
#include <stdio.h>

void makePyramid(int n);
void indent(int n);
void block(int n);


int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 0 || n > 23);
    makePyramid(n);
}

void makePyramid(int n)
{
    for (int i = 0; i < n; i++)
    {
        indent(n - i - 1);
        block(i + 1);
        printf("  ");
        block(i + 1);
        printf("\n");
    }
}

void indent(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

void block(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}

File: mario.py
from cs50 import get_int

# prompt for a non-negative, less-than-23 integer height
while True:
    n = get_int("Height: ")
    if n >= 0 and n <= 23:
        break

# printing
for i in range(n):

    # print spaces
    for j in range(n - (i + 1)):
        print(" ", end="")

    # print blocks
    for k in range(i + 1):
        print("#", end="")

    # print gap in between
    print("  ", end="")

    # print blocks
    for l in range(i + 1):
        print("#", end="")

    # new line
    print()